vault/expiration.go

```diff
@@ -902,6 +935,17 @@ func (m *ExpirationManager) renewAuthEntry(req *logical.Request, le *leaseEntry,
 
 // loadEntry is used to read a lease entry
 func (m *ExpirationManager) loadEntry(leaseID string) (*leaseEntry, error) {
+	restoreMode := m.inRestoreMode()
+	if restoreMode {
+		m.restoreLock.Lock()
+		defer m.restoreLock.Unlock()
```

Comment:
You should update restoreMode again here after grabbing the lock. Otherwise we could have seen that we're in restore mode, go out of restore mode before grabbing the lock, and end up updating pending when it may not be safe to do so anymore.

Reply:
I found that doing the parallel implementation.  I'll address here before merging.

Reply:
This is fixed.